### PR TITLE
MISP V2

### PR DESCRIPTION
**Pull request by jochman**
## Status
Ready

## Related Issues
fixes: 
- https://github.com/demisto/etc/issues/14177
- https://github.com/demisto/etc/issues/13928 (not completed)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Screenshots
Paste here any images that will help the reviewer

branch | PR
misp-python-three | MISP V2

## Required version of Demisto
x.x.x

## Does it break backward compatibility?
   - No

## Must have
- [x] Tests
- [x] Documentation (with link to it)
- [ ] Code Review

## Additional changes
added misp-delete-event command
must be running on pymisp docker: 
- https://github.com/demisto/dockerfiles/tree/master/docker/pymisp
- https://hub.docker.com/r/demisto/pymisp

### CODE REVIEW

File: Scripts/script-misp_download_sample.yml [hunk 1]
```diff
@@ -5,14 +5,26 @@ name: misp_download_sample
...
-  if 'message' in res[0]['Contents'] and res[0]['Contents']['message'] == 'No hits with the given parameters.':
-      demisto.results(res[0]['Contents']['message'])
-  else:
-      filename = res[0]['Contents']['result'][0]['filename']
-      fileContent = base64.b64decode(res[0]['Contents']['result'][0]['base64'])
-      demisto.results(fileResult(filename, fileContent))
+  # Executing command if MISP V2 has an active instance
+  misp_v2 = False
+  for instance_name, data in demisto.getModules().items():
+      if data.get('brand') == "MISP V2" and data.get('state') == 'active':
```
Comment (Itay4) [inline Scripts/script-misp_download_sample.yml]: `"` -> `'`

File: Scripts/script-misp_download_sample.yml [hunk 2]
```diff
@@ -5,14 +5,26 @@ name: misp_download_sample
 script: |+
   import base64
```
Comment (Itay4) [inline Scripts/script-misp_download_sample.yml]: You should add release notes saying that this script is deprecated, and use the new command instead

File: Scripts/script-misp_download_sample.yml [hunk 3]
```diff
@@ -5,14 +5,26 @@ name: misp_download_sample
...
+  for instance_name, data in demisto.getModules().items():
+      if data.get('brand') == "MISP V2" and data.get('state') == 'active':
+          misp_v2 = True
+          break
 
+  if misp_v2:
+      demisto.results(demisto.executeCommand('misp-download-sample', demisto.args()))
+  else:
+      if not get_hash_type(demisto.args().get('hash')) == 'Unknown':
+          res = demisto.executeCommand('internal-misp-download-sample', demisto.args() )
```
Comment (Itay4) [inline Scripts/script-misp_download_sample.yml]: unnecessary white space

File: Scripts/script-misp_download_sample.yml [hunk 4]
```diff
@@ -5,14 +5,26 @@ name: misp_download_sample
...
+          if 'message' in res[0]['Contents'] and res[0]['Contents']['message'] == 'No hits with the given parameters.':
+              demisto.results(res[0]['Contents']['message'])
+          else:
+              filename = res[0]['Contents']['result'][0]['filename']
+              fileContent = base64.b64decode(res[0]['Contents']['result'][0]['base64'])
+              demisto.results(fileResult(filename, fileContent))
+      else:
+          demisto.results('No such hash')
```
Comment (Itay4) [inline Scripts/script-misp_download_sample.yml]: This should be `return_error` and with better error message saying that the hash length is invalid

File: TestPlaybooks/playbook-MISP_V2-Test.yml [hunk 5]
```diff
@@ -0,0 +1,649 @@
...
+    note: false
+    timertriggers: []
+  "5":
+    id: "5"
+    taskid: 47e09105-6f16-4bc8-8cac-efe253b240b1
+    type: title
+    task:
+      id: 47e09105-6f16-4bc8-8cac-efe253b240b1
+      version: -1
+      name: Untitled Task
```
Comment (Itay4) [inline TestPlaybooks/playbook-MISP_V2-Test.yml]: Please name the tasks with indicative name

File: TestPlaybooks/playbook-MISP_V2-Test.yml [hunk 6]
```diff
@@ -0,0 +1,649 @@
...
+    note: false
+    timertriggers: []
+  "1":
+    id: "1"
+    taskid: 3aa71491-43fb-4e1a-8a03-613c19a59e50
+    type: condition
+    task:
+      id: 3aa71491-43fb-4e1a-8a03-613c19a59e50
+      version: -1
+      name: Is MISP V2 active?
```
Comment (Itay4) [inline TestPlaybooks/playbook-MISP_V2-Test.yml]: This task is unnecessary

File: Integrations/integration-MISP_V2.yml [hunk 7]
```diff
@@ -0,0 +1,1334 @@
...
+            md_list = list()
+            for i_event in misp_response:
+                event = i_event['Event']
+                i_event['Event']['RelatedEvent'] = [r_event.get('Event') for r_event in event.get('RelatedEvent')]
+
+            for i_event in misp_response:
+                event = i_event['Event']
+                misp_organisation = f"MISP.{event.get('orgc_name')}"
+                dbot_score = get_dbot_level(event.get('threat_level_id'))
+                # Build RelatedEvent
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: yeap, for further development

File: Integrations/integration-MISP_V2.yml [hunk 8]
```diff
@@ -0,0 +1,1334 @@
...
+                dbot_list.append(dbot_obj)
+                md_list.append(md_obj)
+
+            # Building entry
+            ec = {
+                outputPaths.get('file'): file_list,
+                outputPaths.get('dbotscore'): dbot_list,
+            }
+
+            md = tableToMarkdown(f'Results found in MISP for hash: {file_hash}', md_list)
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: The header is resembling MISP's gui

File: Integrations/integration-MISP_V2.yml [hunk 9]
```diff
@@ -0,0 +1,1334 @@
...
+                'ContentsFormat': formats['json'],
+                'ReadableContentsFormat': formats['markdown'],
+                'HumanReadable': md,
+            })
+
+    def add_tag():
+        uuid = demisto.args().get('uuid')
+        tag = demisto.args().get('tag')
+
+        MISP.tag(uuid, tag)
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: if error occurred the except phrase will catch it

File: Integrations/integration-MISP_V2.yml [hunk 10]
```diff
@@ -0,0 +1,1334 @@
...
+      default: true
+      description: Event name
+    - name: published
+      auto: PREDEFINED
+      predefined:
+      - "false"
+      - "true"
+      description: Publish the event
+      defaultValue: "false"
+    - name: threat_level_id
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: I'm reserving bc, so I don't think changing it to dropdown with threat level names could do so

[the remark this answers is not in the document]
Comment (Itay4) [reply Integrations/integration-MISP_V2.yml]: Why not? Even if you set values, you can still insert any value you want to the argument. You will just need to handle that in the code.

File: Integrations/integration-MISP_V2.yml [hunk 11]
```diff
@@ -0,0 +1,1334 @@
...
+      description: All tags name in the event
+    description: Add attribute to existing MISP event
+  - name: misp-upload-sample
+    arguments:
+    - name: fileEntryID
+      required: true
+      description: the entry ID of the file to upload
+    - name: event_id
+      description: The event's ID to add the upload the file to
+    - name: distribution
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: no, just int number

File: Integrations/integration-MISP_V2.yml [hunk 12]
```diff
@@ -0,0 +1,1334 @@
...
+        marked as "to_ids" or not.
+    - name: category
+      description: 'The category that will be assigned to the uploaded samples. Valid
+        options are: Payload delivery, Artifacts dropped, Payload Installation, External
+        Analysis.'
+    - name: info
+      description: Used to populate the event info field if no event ID supplied.
+        Alternatively, if not set, MISP will simply generate a message showing that
+        it's a malware sample collection generated on the given day.
+    - name: analysis
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: no, theyr'e just numbers

File: Integrations/integration-MISP_V2.yml [hunk 13]
```diff
@@ -0,0 +1,1334 @@
...
+    description: Add attribute to existing MISP event
+  - name: misp-download-sample
+    arguments:
+    - name: hash
+      required: true
+      description: 'A hash in MD5 format. If allSamples is set, this can be any one
+        of the following: md5, sha1, sha256.'
+    - name: eventID
+      description: If set, it will only fetch data from the given event ID.
+    - name: allSamples
```
[the remark this answers is not in the document]
Comment (jochman) [reply Integrations/integration-MISP_V2.yml]: it is necessary. allSamples is do working right now

[the remark this answers is not in the document]
Comment (Itay4) [reply Integrations/integration-MISP_V2.yml]: I didn't say it's not working, but it's useless. You can just make it to accept all types of hashes.

**Comment by ronykoz** [comment]
@jochman seems ok now, that was a server problem

**Pull request merged**